src/python.rs

```diff
@@ -345,6 +532,24 @@ impl Backend {
 		let contents = Cow::from(rope.clone());
 		let contents = contents.as_bytes();
 		let root = some!(top_level_stmt(ast.root_node(), offset));
+
+		// Parse imports from the current file
+		let imports = self.parse_imports(contents).unwrap_or_default();
+		debug!("Parsed imports: {:?}", imports);
+
+		// Check if cursor is on an imported identifier
+		if let Some(cursor_node) = ast.root_node().descendant_for_byte_range(offset, offset)
+			&& cursor_node.kind() == "identifier"
+		{
+			let identifier = String::from_utf8_lossy(&contents[cursor_node.byte_range()]);
+			debug!("Checking identifier '{}' at offset {}", identifier, offset);
+
+			// Try to resolve import location
+			if let Some(location) = self.resolve_import_location(&imports, &identifier, contents)? {
```

Comment:
Seems like we end up comparing with a byte slice, so we can pass `identifier` directly without parsing it back as a string